tests/superset_test_config.py

```diff
@@ -59,3 +64,45 @@ class CeleryConfig(object):
 CELERY_CONFIG = CeleryConfig
 
 RESULTS_BACKEND_USE_MSGPACK = True
+
+
+def DATE(
+    ts: datetime, day_offset: SupportsInt = 0, hour_offset: SupportsInt = 0
+) -> str:
+    """Current day as a string"""
+    day_offset, hour_offset = int(day_offset), int(hour_offset)
+    offset_day = (ts + timedelta(days=day_offset, hours=hour_offset)).date()
+    return str(offset_day)
+
+
+class CustomPrestoTemplateProcessor(PrestoTemplateProcessor):
+    """A custom presto template processor for test."""
+
+    engine = "presto"
+
+    def process_template(self, sql: str, **kwargs) -> str:
+        """Processes a sql template with $-style macro using regex."""
+        # Add custom macros functions.
+        macros = {
+            "DATE": partial(DATE, datetime.utcfromtimestamp(0))
```

Comment:
I would prefer to mock the time function rather than keep it in the macro definition